src/devices/Arduino/samples/Arduino.Monitor.cs

```diff
@@ -0,0 +1,300 @@
+                        modeName = "CPU Load";
+                        disp.Output.ReplaceLine(1, hardwareMonitor.GetCpuLoad().ToString("s1", CultureInfo.CurrentCulture));
+                        break;
+
+                    case 9:
+                        modeName = "Total power dissipation";
+                        var powerSources = hardwareMonitor.GetSensorList().Where(x => x.SensorType == SensorType.Power);
+                        Power totalPower = Power.Zero;
+                        foreach (var power in powerSources)
+                        {
+                            if (power.Name != "CPU Cores" && power.TryGetValue(out Power powerConsumption)) // included in CPU Package
+                            {
+                                totalPower = totalPower + powerConsumption;
+                            }
+                        }
+
+                        disp.Output.ReplaceLine(1, totalPower.ToString("s1", CultureInfo.CurrentCulture));
+                        break;
+
+                    case 10:
+                        modeName = "Energy consumed";
+                        var energySources = hardwareMonitor.GetSensorList().Where(x => x.SensorType == SensorType.Energy);
+                        Energy totalEnergy = Energy.FromWattHours(0); // Set up the desired output unit
+                        foreach (var e in energySources)
+                        {
+                            if (!e.Name.StartsWith("CPU Cores") && e.TryGetValue(out Energy powerConsumption)) // included in CPU Package
+                            {
+                                totalEnergy = totalEnergy + powerConsumption;
+                            }
+                        }
+
+                        disp.Output.ReplaceLine(1, totalEnergy.ToString("s1", CultureInfo.CurrentCulture));
+                        break;
+                }
+
+                int displayWidth = disp.Output.Size.Width;
+                if (modeName.Length > displayWidth)
+                {
+                    // Add one space at the end, makes it a bit easier to read
+                    if (firstCharInText < modeName.Length - displayWidth + 1)
+                    {
+                        firstCharInText++;
+                    }
+                    else
+                    {
+                        firstCharInText = 0;
+                    }
+
+                    disp.Output.ReplaceLine(0, modeName.Substring(firstCharInText));
+                }
+
+                if (modeName != previousModeName)
+                {
+                    disp.Output.ReplaceLine(0, modeName);
+
+                    previousModeName = modeName;
+                    firstCharInText = 0;
+                }
+
+                buttonClicked.WaitOne(sleeptime);
+            }
+
+            hardwareMonitor.Dispose();
+            disp.Output.Clear();
+            disp.Dispose();
+            bmp?.Dispose();
+            gpioController.Dispose();
```

Comment:
1. consider using `using` statement instead
2. Why are you null checking for bmp but not for others?

Reply:
The others are non-nullable. The bmp sensor is optional.